HACKING.md

```diff
@@ -520,6 +520,228 @@ Common instructions for porting:
 
  7. All components should be named following the `UpperCamelCase` convention.
 
+React state management
+----------------------
+
+For simple components with a small amount of isolated state,
+`React.useState` works great.
+
+For complex pages (let's call these apps), where an action in one deeply-
+nested part of the component tree can affect the state in a sibling or
+higher-up part of the tree (e.g. in the relationship editor), it's
+recommended to have one [reducer](https://beta.reactjs.org/learn/extracting-state-logic-into-a-reducer)
+in the very top component and pass all state downward as props.  React calls
+this concept [lifting state up](https://reactjs.org/docs/lifting-state-up.html)
+to create a [top-down data flow](https://reactjs.org/docs/state-and-lifecycle.html#the-data-flows-down).
```

Comment:
The first link says "The new docs will soon replace this site, which will be archived" and points to https://beta.reactjs.org/learn/sharing-state-between-components - not sure if you feel that's better?

Reply:
It felt weird linking to something with beta in the subdomain, but some of the docs only existed on the beta site, so.  I guess I'll just use beta for everything and fix the links later if they break.